File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Ugly Number II.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        num = [1]
        i, j, k = 0, 0, 0
        for _ in range(n):
            l = min(num[i]*2, num[j]*3, num[k]*5)
            num.append(l)
            if l == num[i]*2:
                i += 1
            if l == num[j]*3:
                j += 1
            if l == num[k]*5:
                k += 1
        return num[n-1]

File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Ugly Number II.rs
use std::cmp::min;
impl Solution {
    pub fn nth_ugly_number(n: i32) -> i32 {
        let n = n as usize;
        let mut v = vec![1;n+1];
        let mut i = 0 as usize;
        let mut k = 0 as usize;
        let mut j = 0 as usize;
        for l in 0..n{
            let t = min(v[i]*2, min(v[j]*3, v[k]*5));
            if t == v[i]*2{
                i = i+1;
            }
            if t == v[j]*3{
                j = j+1;
            }
            if t == v[k]*5{
                k = k+1
            }
            v[l+1] = t;
        }
        return v[n-1];
    }
}